File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int start = 0, end = s.length() - 1;

        while (start < end) {
            if (isVowel(s[start]) && isVowel(s[end])) {
                swap(s[start], s[end]);
                start++;
                end--;
            } else if (isVowel(s[start])) {
                end--;
            } else if (isVowel(s[end])) {
                start++;
            } else {
                start++;
                end--;
            }
        }
        return s;
    }

private:
    bool isVowel(char c) {
        return (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' ||
                c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U');
    }
};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        char[] a = s.toCharArray();
        int i = 0; 
        int j = s.length() -1;
        while(i < j) {
            System.out.print(i);
            System.out.println(j);
            if(isVowel(a[i]) && isVowel(a[j])) {
                char temp = a[i];
                a[i] = a[j];
                a[j] = temp;
                i++;
                j--;
            }
            if(!isVowel(a[i])) {
                i++;
            }
            if(!isVowel(a[j])) {
                j--;
            }
        }
        return String.valueOf(a);
    }
    private boolean isVowel(char c) {
        if(c == 'a' || c== 'A' || c== 'e' || c == 'E'|| c == 'i' || c == 'I' || c == 'o' || c == 'O' || c == 'u' || c == 'U') {
            return true;
        }
        return false;
    }
}